vm/vm_test.go

```diff
@@ -362,15 +436,29 @@ func TestVM(t *testing.T) {
 						t.Fatal(err)
 					}
 				}
-				err = vm.Execute()
-				require.Equal(t, testCase.err, err)
+				return vm.Execute()
 			}
 			if len(splitSet) == 0 {
-				verify(nil)
+				err := verify(nil)
+				require.Equal(t, testCase.err, err)
 				return
 			}
-			for _, splitAsset := range splitSet {
-				verify(splitAsset)
+
+			// For splits, sort by ascending value so that we fail
+			// early on invalid zero-value locators.
+			splitAssets := maps.Values(splitSet)
+			sort.Slice(splitAssets, func(i, j int) bool {
+				return splitAssets[i].Asset.Amount <
+					splitAssets[j].Asset.Amount
+			})
+			for _, splitAsset := range splitAssets {
+				err := verify(splitAsset)
+				if err != nil {
```

Comment:
Both if and else do the same assertion? Could this just be `require.Equal(t, testCase.err, err)` and then `if err != nil { return }`?